examples/platforms/emsk/random.c

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright (c) 2016, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements a pseudo-random number generator.
+ *
+ * @warning
+ *   This implementation is not a true random number generator and does @em satisfy the Thread requirements.
+ *
+ */
+
+#include <common/code_utils.hpp>
+
+#include "openthread/platform/random.h"
+#include "openthread/platform/radio.h"
+#include "platform-emsk.h"
+
+#include <stdlib.h>
+
+#include <stdio.h>
+
+static unsigned int seed;
+
+void emskRandomInit(void)
+{
+    unsigned int i;
+
+    printf("Node No.:");
+    scanf("%d", &i);
+    printf("%d\n", i);
+    seed = 10 + i;
+    srand(seed);
```

Comment:
So, the problem is that the Thread Spec requires a true random number generator. (Thread 1.1.1, 7.4.6)  For an example implementation its probably fine to only use a pseudo random number, but I would still prefer if we can seed it with some sort of entropy.

Is there some sort of quasi-random value we can get from the chip?  Even like microseconds since boot, or something like that, just to mix in a bit of entropy so the number sequence isn't always the same.  I unfortunately cant find an actual data sheet on the chip, so I can't really suggest anything.